### PR TITLE
[CL] Option to fix amount of kinetic reactant

### PR DESCRIPTION
**Pull request by joboog**
This pull request adds an option to fix the *amount* of a defined *kinetic_reactant* to its *initial_amount*. Subsequently the change of  the *amount* of a *kinetic_reactant* in time step i does not affect the *amount* in subsequent time steps. 
This is necessary when using a *kinetic_reactant* as reaction rate that only transfers mass between different reactants but does not represent a reactive species itself. 

1. [ ] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.2)
2. [x] Tests covering your feature were added?
3. [x ] Any new feature or behavior change was documented?

### CODE REVIEW

File: ChemistryLib/CreateOutput.cpp [hunk 1]
```diff
@@ -31,8 +31,17 @@ std::unique_ptr<Output> createOutput(
...
+
+    for (auto const& kinetic_reactant : kinetic_reactants){
+        if (kinetic_reactant.item_type == ItemType::KineticReactant &&
+                kinetic_reactant.fix_amount) continue;
+
+        accepted_items.push_back(OutputItem(kinetic_reactant.name,
+                                            kinetic_reactant.item_type));
+    }
+
+    //std::transform(kinetic_reactants.begin(), kinetic_reactants.end(),
```
Comment (endJunction) [inline ChemistryLib/CreateOutput.cpp]: Drop unused code snippets.

File: ChemistryLib/PhreeqcIOData/KineticReactant.h [hunk 2]
```diff
@@ -24,11 +24,13 @@ struct KineticReactant
     KineticReactant(std::string&& name_,
                     std::string&& chemical_formula_,
                     MeshLib::PropertyVector<double>* amount_,
-                    std::vector<double>&& parameters_)
+                    std::vector<double>&& parameters_,
+                    bool&& fix_amount_)
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIOData/KineticReactant.h]: pass booleans by value, please! const.

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+
+## Overview
+
+This scenario describes the transport of multiple solutes through a saturated media involving a complex heterogeneous kinetic reaction network.
+The example model system is a treatment wetland.
+A treatment wetland is an engineered ecosystem that mimics natural occuring microbiological processes to treat wastewater.
+Basically, such a system consists of a basin filled with media (e.g. gravel or sand) and is passed through by wastewater.
+Multiple microbial organisms attached on the media surface and metabolize wastewater pollutants, which drives the cyclin of carbon and nutrients.
+:w
```
Comment (endJunction) [inline web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: vi is my favourite editor too ;)

File: ChemistryLib/PhreeqcIOData/CreateKineticReactant.cpp [hunk 4]
```diff
@@ -48,6 +48,10 @@ std::vector<KineticReactant> createKineticReactants(
             reactant_config.getConfigParameter<std::vector<double>>(
                 "parameters", {});
 
+        bool fix_amount =
```
Comment (endJunction) [inline ChemistryLib/PhreeqcIOData/CreateKineticReactant.cpp]: const

**Comment by renchao-lu** [comment]
Need to discuss. Why not straightly use the initial amount of a kinetic reactant in the rate expression?

File: Documentation/ProjectFile/prj/chemical_system/kinetic_reactants/kinetic_reactant/t_fix_amount.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+specify if the amount of the kinetic reactant should be fixed to the initial amount over the simulation.
+Pssible values: *true* (fixed); *false* (not fixed); default is *false*.
```
Comment (TomFischer) [inline Documentation/ProjectFile/prj/chemical_system/kinetic_reactants/kinetic_reactant/t_fix_amount.md]:
```suggestion
Possible values: *true* (fixed); *false* (not fixed); default is *false*.
```

**Comment by renchao-lu** [comment]
Thanks for your interpretation. Now I understand your motivation. Please add an additional check in the creation process.

File: Documentation/ProjectFile/prj/chemical_system/kinetic_reactants/kinetic_reactant/t_fix_amount.md (hunk 5 above)
Comment (montoyav) [reply Documentation/ProjectFile/prj/chemical_system/kinetic_reactants/kinetic_reactant/t_fix_amount.md]: I take care of looking in detail the doc description (next week) :-)

Comment (joboog) [reply Documentation/ProjectFile/prj/chemical_system/kinetic_reactants/kinetic_reactant/t_fix_amount.md]: @montoyav thanks!

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+
+-----------------------------------------
+
+
+## Results
+
+Initially, influent wastewater pollutants spread through the model domain by advective--dispersive transport (Figure 3).
+The aeration increases dissolved oxygen concentration $Do$ in the entire domain.
+However, after XXX seconds, rapidly growing heterotrophic bacteria $X_H$ starts to consume available $Do$ and influent organic carbon ($S_A$, $S_F$).
```
Comment (TomFischer) [inline web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: How many time is xxx seconds?

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+Initial growth of $X_H$ and the high concentrations of organic carbon sources at the influent region result in an oxygen depletion.
+This triggers the growth of anaerobic fermenting bacteria $X_{FB}$ at the front and pushes $X_H$ downstream were enough $Do$ is available.
+After organic carbon concentration profiles of $S_F$ and $S_A$ stabilized, autotrophic nitrifying bacteria $X_A$ starts to grow by oxidizing available ammonia nitrogen ($S_{NH}$) to oxidized nitrogen $S_{NO}$.
+After XX seconds (20 days) the wetland has reached a quasi steady--state.
```
Comment (TomFischer) [inline web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: The storage S equals to zero. I would expect that the steady-state reached in the first time step. Is my understanding wrong?

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+-----------------------------------------
+
+|Parameter | Description | Value | Unit |
+|:-------- | :---------- | :---- | :--- |
+| Influent & effluent zone |||
+| $\phi$   | Porosity    | 0.38  | - |
+| $\kappa$ | Permeability | 1.0e-7 | m^2^ |
+| $S$   | Storage | 0.0     | - |
+| $a_L$ | long. Dispersion length | 0.45   | m |
```
Comment (TomFischer) [inline web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: Why is the transversal dispersion length not mentioned?

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc (hunk 7 above)
Comment (joboog) [reply web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: I will have to clarify that I meant the state of the microbial reaction network, not the hydrologic system. Of course, water flow reaches steady--state in the first time step.

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc (hunk 8 above)
Comment (joboog) [reply web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: As it is a 1D model there is only longitudinal dispersion.

**Comment by renchao-lu** [comment]
Looks good to me. Can be merged after documentation review done by Vanessa. @montoyav

**Comment by montoyav** [comment]
In general some additional points in the web description should be corrected/checked: a) Nomenclature should be consistent, b) reference to Samsó is missing, c) Figure with the kinetic reactions was not appearing, d) Figure 3 appears as Fig. 2, e) it is difficult to see from the animated figure which is the t = 0, maybe it will help to see how the time is progressing? f) it is mentioned that in the Figure it is shown the first 4 time steps, but it is not indicated how much time is this, g) simulation time was also not mentioned (20 days?).

File: web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc [hunk 9]
```diff
@@ -121,3 +117,5 @@ Please note that due to the long computation time of the simulation, the corresp
 Boog, J., Kalbacher, T., Nivala, J., Forquet, N., van Afferden, M., Müller, R.A., 2019. Modeling the relationship of aeration, oxygen transfer and treatment performance in aerated horizontal flow treatment wetlands. Water Research. 157 , 321 - 334. https://doi.org/10.1016/j.watres.2019.03.062.
 
 Langergraber, G., Rousseau, D.P.L., García, J., Mena, J., 2009. CWM1: A general model to describe biokinetic processes in subsurface flow constructed wetlands. Water Science and Technology, 59 (9), 1687-1697. https://doi.org/10.2166/wst.2009.131.
+
+Samso...
```
Comment (endJunction) [inline web/content/docs/benchmarks/reactive-transport/wetland/Wetland.pandoc]: ?

**Comment by joboog** [comment]
[release note page](https://github.com/ufz/ogs/wiki/Release-notes-6.2.2) updated

**Pull request merged**